homeassistant/components/webhook/__init__.py

```diff
@@ -90,16 +104,18 @@ async def async_handle_webhook(
     """Handle a webhook."""
     handlers: dict[str, dict[str, Any]] = hass.data.setdefault(DOMAIN, {})
 
+    content_stream: StreamReader | MockStreamReader
+    if isinstance(request, MockRequest):
+        received_from = request.mock_source
+        content_stream = request.content
+        method_name = request.method
+    else:
+        received_from = request.remote
+        content_stream = request.content
+        method_name = request.method
```

Comment:
```suggestion
    content_stream = request.content
    method_name = request.method
    if isinstance(request, MockRequest):
        received_from = request.mock_source
    else:
        received_from = request.remote
```

Reply:
I've tried this a few times and keep stumbling with the mypy checks. Do you have a suggestion on how to make this work?

```
mypy homeassistant/components/webhook
homeassistant/components/webhook/__init__.py:108: error: Argument 1 to "__get__" of "reify" has incompatible type "Union[Request, MockRequest]"; expected "_TSelf[StreamReader]"  [arg-type]
homeassistant/components/webhook/__init__.py:109: error: Argument 1 to "__get__" of "reify" has incompatible type "Union[Request, MockRequest]"; expected "_TSelf[str]"  [arg-type]
Found 2 errors in 1 file (checked 2 source files)
```
https://github.com/aio-libs/aiohttp/blob/ae7703cefd1f8c8ad02bfc21cdc92c367f2666b9/aiohttp/web_request.py#L613

Reply:
Ugh, lets keep it as is :)